### PR TITLE
Improve error messages

### PR DESCRIPTION
**Pull request by dcodeIO**
This PR attempts to replace all `Operation not supported` errors with better messages eventually. Turns out that doing this properly is more work than expected since it also involves quite a bit of refactoring.

- [x] Update remaining errors

### CODE REVIEW

**Comment by dcodeIO** [comment]
Alright, `Operation not supported` is gone now and has been replaced with various new and existing error messages, with `Not implemented` becoming the new thing. The difference to "Operation not supported" is that it's really just "not implemented" (yet) even though it should be / is planned.

**Comment by dcodeIO** [comment]
Can probably go on forever with refactoring here, but the initial goal is achieved so I can as well stop here I assume :)

File: src/builtins.ts [hunk 1]
```diff
@@ -1884,16 +1747,26 @@ export function compileCall(
     case BuiltinSymbols.atomic_or:
     case BuiltinSymbols.atomic_xor:
     case BuiltinSymbols.atomic_xchg: {
-      if (!compiler.options.hasFeature(Feature.THREADS)) break;
       if (
+        checkFeatureEnabled(Feature.THREADS, reportNode, compiler) |
         checkTypeRequired(typeArguments, reportNode, compiler, true) |
         checkArgsOptional(operands, 2, 3, reportNode, compiler)
```
Comment (MaxGraey) [inline src/builtins.ts]: Could you also add checking type argument is not float (only integers) for atomics?

Comment (dcodeIO) [reply src/builtins.ts]: It should do that afterwards, that is emit an `Operation X cannot be applied to type Y` error if there's no matching case. https://github.com/AssemblyScript/assemblyscript/pull/869/files/c01a9c63ffd1d25a24a38cf7eea6c0c7f3f74cea#diff-2c8ebd9fe94e61d77bc89956497d80daR1768

Comment (MaxGraey) [reply src/builtins.ts]: Right. [Now I see](https://github.com/AssemblyScript/assemblyscript/pull/869/files/c01a9c63ffd1d25a24a38cf7eea6c0c7f3f74cea#diff-2c8ebd9fe94e61d77bc89956497d80daR1756)

**Pull request merged**